app/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveAuth.scala

```diff
@@ -63,6 +63,29 @@ class HelpToSaveAuth(frontendAuthConnector: FrontendAuthConnector, metrics: Metr
         }
     }
 
+  def authorisedForHtsWithNINOAndName(action: HtsAction[HtsContextWithNINOAndName])(redirectOnLoginURL: String): Action[AnyContent] =
+    Action.async { implicit request ⇒
+      val timer = metrics.authTimer.time()
+
+      authorised(AuthWithCL200)
+        .retrieve(Retrievals.name and Retrievals.itmpName and authorisedEnrolments) {
+          case maybeName ~ maybeItmpName ~ authorisedEnrols ⇒
+            val time = timer.stop()
+
+            withNINO(authorisedEnrols.enrolments, time){ nino ⇒
+              val name: Option[(String, String)] = for {
+                forename ← maybeName.name.orElse(maybeItmpName.givenName)
```

Comment:
do you think we can give preference to the `itmpNames` first, then look into user details name ? or it doesnt matter ?

Reply:
I don't know what the preference should be, should probably make sure